activerecord/lib/active_record/reflection.rb

```diff
@@ -618,6 +622,30 @@ def derive_join_table
         def primary_key(klass)
           klass.primary_key || raise(UnknownPrimaryKey.new(klass))
         end
+
+        # Finds inverse reflection from +klass+ with properties equal to options with provided keys.
+        def find_inverse_reflection_by_options keys
+          result = klass.reflections.find do |name, refl|
```

Comment:
here is a loop through all of models relation, do I need to write benchmarks for it?